src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FootnoteService.cs

```diff
@@ -60,83 +56,70 @@ public class FootnoteService : IFootnoteService
-                .OnSuccess(async footnotes => 
-                    await footnotes.ForEachAsync(async footnote =>
-                    {
-                        var filters = footnote.Filters
-                            .Select(filterFootnote => filterFootnote.FilterId).ToList();
-                        var filterGroups = footnote.FilterGroups
-                            .Select(filterGroupFootnote => filterGroupFootnote.FilterGroupId).ToList();
-                        var filterItems = footnote.FilterItems
-                            .Select(filterItemFootnote => filterItemFootnote.FilterItemId).ToList();
-                        var indicators = footnote.Indicators
-                            .Select(indicatorFootnote => indicatorFootnote.IndicatorId).ToList();
-                        var subjects = footnote.Subjects
-                            .Select(subjectFootnote => subjectFootnote.SubjectId).ToList();
-
-                        return await CreateFootnote(destinationReleaseId,
-                            footnote.Content,
-                            filters,
-                            filterGroups,
-                            filterItems,
-                            indicators,
-                            subjects);
-                    }));
+            return await _contentPersistenceHelper.CheckEntityExists<Release>(destinationReleaseId)
+                .OnSuccessDo(_userService.CheckCanUpdateRelease)
+                .OnSuccess(() => GetFootnotes(sourceReleaseId))
+                .OnSuccess(async footnotes =>
+                {
+                    return await footnotes
+                        .ToAsyncEnumerable()
+                        .SelectAwait(async footnote =>
+                        {
+                            var filters = footnote.Filters
+                                .Select(filterFootnote => filterFootnote.FilterId).ToList();
+                            var filterGroups = footnote.FilterGroups
+                                .Select(filterGroupFootnote => filterGroupFootnote.FilterGroupId).ToList();
+                            var filterItems = footnote.FilterItems
+                                .Select(filterItemFootnote => filterItemFootnote.FilterItemId).ToList();
+                            var indicators = footnote.Indicators
+                                .Select(indicatorFootnote => indicatorFootnote.IndicatorId).ToList();
+                            var subjects = footnote.Subjects
+                                .Select(subjectFootnote => subjectFootnote.SubjectId).ToList();
+
+                            return await _footnoteRepository.CreateFootnote(destinationReleaseId,
+                                footnote.Content,
+                                filters,
+                                filterGroups,
+                                filterItems,
+                                indicators,
+                                subjects);
```

Comment:
Same as above - would suggest using named parameters to avoid mixing up any of the id lists by accident.